### ISSUE TITLE
"exclude" shouldn't exclude filename that only match using startsWith

### ISSUE THREAD

**Issue by bsvensson** [jsdoc3/jsdoc#979]
Currently, if the exclude in the conf file include a filename like `somefolder/MyModule.js`, then anything that startsWith that with that will be excluded.  This means that if you have the jsdoc in `somefolder/MyModule.jsdoc` that it will not even be parsed.

See comment by @dasa on the https://github.com/jsdoc3/jsdoc/pull/617#issuecomment-90639620 closed pull request.

I think that it makes sense to exclude based on startsWith for folders, but for filenames I think the match should be exact.

Workaround: name the files differently, for example `somefolder/MyModule.doc` or `somefolder/MyModule-doc.js`.

Example conf (where I would expect the jsdoc file to be parsed):
```
{
    "source": {
        "includePattern": ".+\\.js(doc)?$",
        "include": [
            "somefolder/MyModule.jsdoc"
        ],
        "exclude": [
            "somefolder/MyModule.js"
        ]
    }
}
```
Running jsdoc with this conf will show `There are no input files to process.`

**Comment by hegemonic** [comment]
I agree that the current behavior is wrong, although fixing it may be a little tricky.
